### PR TITLE
76816 create webjob skeleton

### PR DESCRIPTION
**Pull request by sakshi15525**
1. Created WebJob Skeleton
2. Added Serilog, App Insights and event hub logging configuration
3. Added Keyvault configuration

### CODE REVIEW

File: src/UKHO.ERPFacade.WebJob/Program.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+using UKHO.ERPFacade.Common.Configuration;
+using UKHO.Logging.EventHubLogProvider;
+
+namespace UKHO.ERPFacade.WebJob
+{
+    [ExcludeFromCodeCoverage]
+    public static class Program
+    {
+        private static readonly InMemoryChannel s_aIChannel = new();
+        private static readonly string s_assemblyVersion = Assembly.GetExecutingAssembly().GetCustomAttributes<AssemblyFileVersionAttribute>().Single().Version;
```
Comment (jrippington) [inline src/UKHO.ERPFacade.WebJob/Program.cs:21]: Naming standards - rename to WebJobAssemblyVersion or something similar?

**Review by jrippington — CHANGES_REQUESTED** [review]

File: src/UKHO.ERPFacade.WebJob/Program.cs (hunk 1 above)
Comment (sakshi15525) [reply src/UKHO.ERPFacade.WebJob/Program.cs]: Renamed to WebJobAssemblyVersion

File: src/UKHO.ERPFacade.WebJob/Program.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+using System.Reflection;
+using UKHO.ERPFacade.Common.Configuration;
+using UKHO.Logging.EventHubLogProvider;
+
+namespace UKHO.ERPFacade.WebJob
+{
+    [ExcludeFromCodeCoverage]
+    public static class Program
+    {
+        private static readonly InMemoryChannel s_aIChannel = new();
```
[the remark this answers is not in the document]
Comment (sakshi15525) [reply src/UKHO.ERPFacade.WebJob/Program.cs]: Renamed to TelemetryChannel

File: src/UKHO.ERPFacade.WebJob/Program.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+#endif
+            serviceCollection.AddLogging(loggingBuilder =>
+            {
+                loggingBuilder.AddConfiguration(configuration.GetSection("Logging"));
+                loggingBuilder.AddConsole();
+                loggingBuilder.AddDebug();
+                loggingBuilder.AddSerilog();
+                loggingBuilder.AddAzureWebAppDiagnostics();
+
+                var eventHubConfig = configuration.GetSection("EventHubLoggingConfiguration").Get<EventHubLoggingConfiguration>();
```
Comment (Shirin14926) [inline src/UKHO.ERPFacade.WebJob/Program.cs:109]: use explicit instead of var

**Review by Vishal14583 — APPROVED** [review]
LGTM

File: src/UKHO.ERPFacade.WebJob/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  "KeyVaultSettings": {
+    "ServiceUri": ""
+  },
+  "ApplicationInsights": {
+    "ConnectionString": ""
+  },
+  "EventHubLoggingConfiguration": {
+    "MinimumLoggingLevel": "Warning",
+    "UkhoMinimumLoggingLevel": "Information",
+    "Environment": "Dev",
```
[the remark this answers is not in the document]
Comment (Vishal14583) [reply src/UKHO.ERPFacade.WebJob/appsettings.json:22]: Removed

**Review by Shirin14926 — APPROVED** [review]
Thanks @Vishal14583 and @sakshi15525 for fixing minor comments. LGTM

File: src/UKHO.ERPFacade.Common/Helpers/AzureTableReaderWriter.cs
Comment (jrippington) [inline src/UKHO.ERPFacade.Common/Helpers/AzureTableReaderWriter.cs:1]: Folder name should be changed from 'Helpers' to 'IO' to match the namespace

**Review by jrippington — CHANGES_REQUESTED** [review]

File: src/UKHO.ERPFacade.Common/Helpers/AzureTableReaderWriter.cs
Comment (Vishal14583) [reply src/UKHO.ERPFacade.Common/Helpers/AzureTableReaderWriter.cs:1]: updated

**Review by jrippington — APPROVED** [review]
Looks good!

**Review by JiviteshT — APPROVED** [review]
Approved to progress merge

**Pull request merged**